### PR TITLE
Make it easier to override the Forms Page presenter

### PR DESCRIPTION
**Pull request by martijn00**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

Feature

### :arrow_heading_down: What is the current behavior?

It is hard to use a custom Page presenter on Forms

### :new: What is the new behavior (if this is a feature change)?

Easier to override

### :boom: Does this PR introduce a breaking change?

No

### :memo: Links to relevant issues/docs

Fixes https://github.com/MvvmCross/MvvmCross/issues/2802

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [x] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [x] Rebased onto current develop

### CODE REVIEW

File: MvvmCross.Forms/Core/IMvxFormsSetup.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (nickrandolph) [inline MvvmCross.Forms/Core/IMvxFormsSetup.cs]: Can we remove any of these?

Comment (martijn00) [reply MvvmCross.Forms/Core/IMvxFormsSetup.cs]: @nickrandolph done

File: MvvmCross.Forms/Platforms/Android/Core/MvxFormsAndroidSetup.cs [hunk 2]
```diff
@@ -69,10 +75,18 @@ public virtual Application FormsApplication
 
         protected abstract Application CreateFormsApplication();
 
+        protected virtual IMvxFormsPagePresenter CreateFormsPagePresenter(IMvxFormsViewPresenter viewPresenter)
```
Comment (nickrandolph) [inline MvvmCross.Forms/Platforms/Android/Core/MvxFormsAndroidSetup.cs]: Do we need to remove the code in the MvxFormsAndroidViewPresenter that creates the MvxFormsPagePresenter?

File: MvvmCross.Forms/Platforms/Uap/Core/MvxFormsWindowsSetup.cs [hunk 3]
```diff
@@ -41,12 +42,43 @@ public virtual Application FormsApplication
...
+        protected override void InitializeIoC()
+        {
+            base.InitializeIoC();
+            Mvx.RegisterSingleton<IMvxFormsSetup>(this);
+        }
+
+        protected override void InitializeApp(IMvxPluginManager pluginManager, IMvxApplication app)
+        {
+            base.InitializeApp(pluginManager, app);
+            _viewAssemblies.AddRange(GetViewModelAssemblies());
```
Comment (nickrandolph) [inline MvvmCross.Forms/Platforms/Uap/Core/MvxFormsWindowsSetup.cs]: Why is it necessary for all the viewmodel assemblies to be added into the viewassemblies? If the forms Application class is in the same assembly as the viewmodels, the assembly will already be in the viewassemblies. This override shouldn't be necessary for any platform now

Comment (martijn00) [reply MvvmCross.Forms/Platforms/Uap/Core/MvxFormsWindowsSetup.cs]: This is when you are not using the generic. Maybe we've got to look into cleanup or doing it another way, but that's for another PR.

**Pull request merged**